database/DatabaseExampleSwiftUI/DatabaseExample/Shared/Models/Post.swift

```diff
@@ -47,9 +49,33 @@ struct Post: Identifiable {
     self.title = title
     self.body = body
     self.starCount = starCount
+    self.starDictionary = [:]
   }
 
   init() {
     self.init(id: "", uid: "", author: "", title: "", body: "")
   }
+
+  mutating func didTapStarButton(isStarred: Bool) {
+    let ref: DatabaseReference = {
+      Database.database().reference()
+    }()
+    if isStarred {
+      // add current user to the post.starDictionary
+      starDictionary[uid] = true
+      ref.child("/posts/\(id)/starDictionary").setValue(starDictionary)
+      ref.child("/user-posts/\(uid)/\(id)/starDictionary").setValue(starDictionary)
```

Comment:
nit: I cannot see the entire data schema but looks like we duplicate a post in two collections. I think we should rather have a collection of post IDs for a user and fetch the posts by ID from the post collection to avoid duplication.

Reply:
Sounds good. I started the fetching and posting data structure by following the legacy example. Since SwiftUI has a slight different implementation, I will create a design doc that talks about my intended design structure for the app.

Reply:
Ah, I didn't know we'd had this data structure already. Then, I think, we can postpone modifying it and take it as a separate project and follow the existing approach so far.
Though to make the future refactoring easier we should isolate the Firebase Database logic and expose as little implementation details as possible to the rest of the app. I would suggest making `starDictionary` private and expose methods like `(un)startByUser(userID: String)` and `isStarredByUser(userID: String)`. In this case we will be able to change the implementation in the future without changing code that uses `Post` model.